core/src/processing/core/PApplet.java

```diff
@@ -10105,6 +10105,8 @@ static public void runSketch(final String[] args,
     sketch.present = present;
     sketch.fullScreen = fullScreen;
 
+    sketch.pixelDensity = sketch.displayDensity();
```

Comment:
I stumbled across this just now while helping a new CS teacher come up to speed on one of my former class projects. The project uses a nested loop and the width and height system variables to change the color of each pixel in the pixels array (as described in chapter 15 of Shiffman's "Learning Processing"). As of this change, that same loop only modifies the pixels in a small portion of the display window on my high DPI laptop. It's a nasty little bug if you haven't faithfully kept up with the Processing release notes. Perhaps you could display a warning message to ease folks into it?

Reply:
Hi @agreyson Thanks for your comment! We made this change because sketches looked blurry on high-DPI displays, but as you found out, it can break older code that works with pixels directly. This happens because different pixelDensity values introduce inconsistencies in per-pixel operations. @catilac and @tychedelia are exploring solutions here: https://github.com/processing/processing4/pull/1145, and suggestions are welcome on the issue thread: https://github.com/processing/processing4/issues/1131. Since the fix is more complex than expected, adding a warning message could be a helpful interim step to prevent the headaches you ran into. Thanks for your suggestion.